controllers/replace_misconfigured_pods.go

```diff
@@ -89,6 +89,25 @@ func (c ReplaceMisconfiguredPods) Reconcile(r *FoundationDBClusterReconciler, co
 			return &Requeue{Error: err}
 		}
 
+		if pvc.Name != desiredPVC.Name {
```

Comment:
Can you please write test cases for that? I also think it makes sense to move everything in the PVC for loop into a private method (that should make the testing easier, similar to `instanceNeedsRemoval`.